### PR TITLE
[One Plans Page][Pricing Cards Block] Bring pricing-cards into one-plans

### PR DESCRIPTION
**Pull request by JingleH**
Put all required blocks together to get an initial page look. Note that this block will still be refactored later into using grids for better maneuverability on responsiveness.

Resolves: https://jira.corp.adobe.com/browse/MWPW-141070

Test URLs:
- Before: https://one-plans--express--adobecom.hlx.page/drafts/jinglhua/pricing-cards?martech=off
- After: https://pricing-cards--express--adobecom.hlx.page/drafts/jinglhua/pricing-cards?martech=off

### CODE REVIEW

File: express/blocks/pricing-cards/pricing-cards.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  );
+  const cards = Array.from(divs[0]).map((_, index) =>
+    blockKeys.reduce((obj, key, keyIndex) => {
+      obj[key] = divs[keyIndex][index];
+      return obj;
+    }, {}),
+  );
+  const disclaimer = el.querySelector(':scope > div:last-child');
+  disclaimer.classList.add('disclaimer');
+  el.querySelectorAll(':scope > div').forEach((div) => div.remove());
```
Comment (vhargrave) [inline express/blocks/pricing-cards/pricing-cards.js:94]: if you just select for every div two lines below that is not the last child then you don't even need to add select the disclaimer, give it a class and add it back again. 
`el.querySelectorAll(':scope > div:not(:last-child)')`

**Review by vhargrave — COMMENTED** [review]
I'm excited for this PR. That performance score you're getting is juicy man! 
Found a few small nits if you're interested. Also not sure if it's on your radar, so just mentioning to 
1. add documentation
2. either add a task to create unit tests or add at least one :P

File: express/blocks/pricing-cards/pricing-cards.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/* eslint-disable object-curly-newline */
```
[the remark this answers is not in the document]
Comment (JingleH) [reply express/blocks/pricing-cards/pricing-cards.js:1]: my good little prettier has conflicting rules with our linting rules 😢 so I always disable all conflicting rules during development. Before final merge or release, I'll bring those rules back and make my coding styles compliant

File: express/blocks/pricing-cards/pricing-cards.css [hunk 3]
```diff
@@ -0,0 +1,215 @@
+:root {
+  --card-width: 319px;
+  --card-padding: 16px 12px;
+}
+main .pricing-cards-container > div {
```
[the remark this answers is not in the document]
Comment (JingleH) [reply express/blocks/pricing-cards/pricing-cards.css]: yeah the reason is that our styles.css would have prevalence over this max-width if our selectors level haven't reached 3, so adding a main was the simplest approach. Good catch with fragments! I've replaced it with .pricing-cards-container > div.pricing-cards-wrapper

File: express/blocks/pricing-cards/pricing-cards.css [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  display: none;
+}
+
+.pricing-cards .card-offer {
+  position: absolute;
+  right: 8px;
+  top: -10px;
+  border-radius: 4px;
+  background: #D3D5FF;
+  padding: 0px 6px 1px 6px;
```
[the remark this answers is not in the document]
Comment (JingleH) [reply express/blocks/pricing-cards/pricing-cards.css]: good one!

**Comment by JingleH** [comment]
> I'm excited for this PR. That performance score you're getting is juicy man! Found a few small nits if you're interested. Also not sure if it's on your radar, so just mentioning to
> 
> 1. add documentation
> 2. either add a task to create unit tests or add at least one :P

yeah the block is not really finished yet. It's using flexboxes but I'll change it into grids, so there will be some major overhaul happening. This is just a PR for getting all required blocks to sit on the same page on a branch and then we can get some initial feedbacks. Planning to do all those followups next week. But if we don't need that initial feedbacks I'll close this. Thanks for pointing these out!

**Pull request merged**